test/integration/test_superdarn.py

```diff
@@ -0,0 +1,902 @@
+        """
+        with bz2.open(map_stream) as fp:
+            dmap_stream = fp.read()
+        dmap = pydarn.DmapRead(dmap_stream, True)
+        dmap_stream_data = dmap.read_records()
+        dmap_write = pydarn.DarnWrite(dmap_stream_data)
+        dmap_write.write_map("test_map.map")
+        dmap = pydarn.DarnRead("test_map.map")
+        dmap_data = dmap.read_map()
+        self.dmap_compare(dmap_stream_data, dmap_data)
+        os.remove("test_map.map")
+
+    def test_DmapWrite_stream_DarnRead_map(self):
+        """
+        Test DmapWrite to write to a stream and have DarnRead
+        the map stream
+        """
+        map_data = copy.deepcopy(map_data_sets.map_data)
+        map_write = pydarn.DmapWrite()
+        map_stream = map_write.write_dmap_stream(map_data)
+
+        map_read = pydarn.DarnRead(map_stream, True)
+        map_read_data = map_read.read_map()
+        self.dmap_compare(map_read_data, map_data)
+
+    def test_DmapWrite_missing_DarnRead_map(self):
+        """
+        Test DmapWrite writes a fitacf file missing the field nave in record 2
+        and DarnRead reads the file
+
+        Behaviour: Raise SuperDARNFieldMissingError
+        """
+        map_missing_field = copy.deepcopy(map_data_sets.map_data)
+        del map_missing_field[0]['stid']
+        dmap_write = pydarn.DmapWrite(map_missing_field)
+        dmap_write.write_dmap("test_missing_map.map")
+
+        darn_read = pydarn.DarnRead("test_missing_map.map")
+        try:
+            darn_read.read_map()
+        except pydarn.superdarn_exceptions.SuperDARNFieldMissingError as err:
+            self.assertEqual(err.fields, {'stid'})
+            self.assertEqual(err.record_number, 0)
+
+        os.remove("test_missing_map.map")
+
+    def test_DmapWrite_extra_DarnRead_map(self):
+        """
+        Test DmapWrite writes a map file with an extra field and DarnRead
+        reads the file
+
+        Behaviour: Raised SuperDARNExtraFieldError
+        """
+        map_extra_field = copy.deepcopy(map_data_sets.map_data)
+        map_extra_field[1].update({'dummy': pydarn.DmapScalar('dummy',
+                                                                 'nothing',
+                                                                 9, 's')})
+        map_extra_field[1].move_to_end('dummy', last=False)
+        dmap_write = pydarn.DmapWrite(map_extra_field, )
+        dmap_write.write_dmap("test_extra_map.map")
+
+        darn_read = pydarn.DarnRead("test_extra_map.map")
+        try:
+            darn_read.read_map()
+        except pydarn.superdarn_exceptions.SuperDARNExtraFieldError as err:
+            self.assertEqual(err.fields, {'dummy'})
+            self.assertEqual(err.record_number, 1)
+        os.remove("test_extra_map.map")
+
+
+if __name__ == '__main__':
+    """
+    Runs the above class in a unittest system.
+    Roughly takes 467 seconds.
+    """
+    pydarn_logger.info("Starting DMAP testing")
+
+    unittest.main()
```

Comment:
Should we maybe a test  in here for RST compatibility?

Reply:
We could but that would definitely screw up `python test` since that would mean someone has to install RST. Maybe that is a good suggestion to ask reviewers to do.

Reply:
I think that will work.